test/validateAllNetworks.test.ts

```diff
@@ -0,0 +1,205 @@
+import { config } from 'dotenv';
+config();
+import {
+    Address,
+    createTestClient,
+    http,
+    parseEther,
+    publicActions,
+    walletActions,
+    TestActions,
+    Hex,
+} from 'viem';
+import {
+    CHAINS,
+    ChainId,
+    SwapKind,
+    Swap,
+    PERMIT2,
+    PublicWalletClient,
+} from '@/index';
+import { Path } from '@/entities/swap/paths/types';
+import { AddressProvider } from '@/entities/inputValidator/utils/addressProvider';
+import {
+    approveSpenderOnTokens,
+    approveTokens,
+    setTokenBalances,
+} from 'test/lib/utils/helper';
+import { POOLS, TOKENS, TestToken, TestPool } from 'test/lib/utils/addresses';
+import { assertSwapExactIn } from 'test/lib/utils/swapHelpers';
+import {
+    ANVIL_NETWORKS,
+    startFork,
+    NetworkSetup,
+} from 'test/anvil/anvil-global-setup';
+
+const protocolVersion = 3;
+
+// Manual mapping between ChainId and ANVIL_NETWORKS keys
+const CHAIN_ANVIL_MAP: Partial<Record<number, keyof typeof ANVIL_NETWORKS>> = {
+    [ChainId.MAINNET]: 'MAINNET',
+    [ChainId.POLYGON]: 'POLYGON',
+    [ChainId.FANTOM]: 'FANTOM',
+    [ChainId.SEPOLIA]: 'SEPOLIA',
+    [ChainId.OPTIMISM]: 'OPTIMISM',
+    [ChainId.MODE]: 'MODE',
+    [ChainId.FRAXTAL]: 'FRAXTAL',
+    [ChainId.AVALANCHE]: 'AVALANCHE',
+    [ChainId.ARBITRUM_ONE]: 'ARBITRUM_ONE',
+    [ChainId.GNOSIS_CHAIN]: 'GNOSIS_CHAIN',
+    [ChainId.SONIC]: 'SONIC',
+    [ChainId.HYPER_EVM]: 'HYPER_EVM',
+    // Add/remove mappings as needed for your test coverage
+};
+
+// Define a minimal config per network for tokens/pool/path
+const NETWORK_CONFIG: Partial<
+    Record<
+        number,
+        {
+            tokenInKey: string;
+            tokenOutKey: string;
+            poolKey: string;
+            inputAmountRaw: bigint;
+        }
+    >
+> = {
+    [ChainId.SEPOLIA]: {
+        tokenInKey: 'BAL',
+        tokenOutKey: 'WETH',
+        poolKey: 'MOCK_WETH_BAL_POOL',
+        inputAmountRaw: 100000000000n,
+    },
+    [ChainId.HYPER_EVM]: {
+        tokenInKey: 'feWETH',
+        tokenOutKey: 'feUSD',
+        poolKey: 'MOCK_FEUSD_FEWETH_POOL',
+        inputAmountRaw: 100000000000n,
+    },
+};
+
+// Optionally override fork block numbers for specific chains. Useful to select pools deployed on V3 after
+// the specified block in anvil-global-setup.ts
+const BLOCK_NUMBER_OVERRIDES: Partial<Record<number, bigint>> = {
+    [ChainId.HYPER_EVM]: 6892528n,
+    [ChainId.MAINNET]: 22788192n,
+};
+
+// List of ChainIds to run the test for. Modify this array to select which chains to test.
+const CHAINS_TO_TEST: number[] = [ChainId.SEPOLIA, ChainId.HYPER_EVM];
+
+describe('validateAllNetworks', () => {
```

Comment:
What is the purpose of these tests? Are you validating that each network has proper configs for testing?

Reply:
looks like purpose is to execute a swap on each network included in `CHAINS_TO_TEST`